front/components/assistant/conversation/input_bar/InputBar.tsx

```diff
@@ -144,68 +148,80 @@ export function AssistantInputBar({
       configurationId: m.id,
     }));
 
-    let contentFragment:
-      | {
-          title: string;
-          content: string;
-          file: File;
-          contentType: string;
-        }
-      | undefined = undefined;
-    if (contentFragmentData) {
-      contentFragment = {
-        ...contentFragmentData,
-        contentType: "file_attachment",
-      };
-    }
-    onSubmit(text, mentions, contentFragment);
+    onSubmit(
+      text,
+      mentions,
+      contentFragmentData.map((cf) => {
+        return {
+          title: cf.title,
+          content: cf.content,
+          file: cf.file,
+        };
+      })
+    );
     resetEditorText();
-    setContentFragmentData(undefined);
+    setContentFragmentData([]);
   };
 
   const onInputFileChange: InputBarContainerProps["onInputFileChange"] =
     useCallback(
       async (event) => {
-        const file = (event?.target as HTMLInputElement)?.files?.[0];
-        if (!file) {
-          return;
-        }
-        if (file.size > 100_000_000) {
-          // Even though we don't display the file size limit in the UI, we still check it here to prevent
-          // processing files that are too large.
-          sendNotification({
-            type: "error",
-            title: "File too large.",
-            description:
-              "PDF uploads are limited to 100Mb per file. Please consider uploading a smaller file.",
-          });
-          return;
-        }
-        const res = await handleFileUploadToText(file);
+        let totalSize = 0;
+        for (const file of (event?.target as HTMLInputElement)?.files || []) {
+          if (!file) {
+            return;
+          }
+          if (file.size > 100_000_000) {
```

Comment:
Shouldn't we compute the total files size as well?

Reply:
No, this limit is totally arbitrary, so checking each file one by one is fine i think.
What really matters are the total extracted characters.